src/string.cr

```diff
@@ -4428,6 +4428,138 @@ class String
       value
     end
   end
+
+  # Implementation of string interpolation of a single string.
+  #
+  # For example, this code will end up invoking this method:
+  #
+  # ```
+  # value = "hello"
+  # "#{value}" # same as String.interpolation(value)
+  # ```
+  #
+  # In this case the implementation just returns a new string with the contents
+  # of `value`.
+  #
+  # NOTE: there should never be a need to call this method instead of using string interpolation.
+  def self.interpolation(value : String)
+    String.new(value.bytesize) do |buffer|
+      buffer.copy_from(value.to_unsafe, value.bytesize)
+      {value.bytesize, value.size_known? ? value.size : 0}
+    end
+  end
+
+  # Implementation of string interpolation of a single non-string value.
+  #
+  # For example, this code will end up invoking this method:
+  #
+  # ```
+  # value = 123
+  # "#{value}" # same as String.interpolation(value)
+  # ```
+  #
+  # In this case the implementation just returns the result of calling `value.to_s`.
+  #
+  # NOTE: there should never be a need to call this method instead of using string interpolation.
+  def self.interpolation(value)
+    value.to_s
```

Comment:
There is no guarantee that it will create a new string, this would be inconsistent with what we can expect I think